examples/pytorch/nanogpt/train.py

```diff
@@ -377,12 +377,15 @@ def train():
                         )
                     iter_num += 1
                     local_iter_num += 1
-
+                    if iter_num % args.checkpoint_step == 0:
+                        ckpt_manager.save(epoch, iter_num)
                     # Termination conditions
                     if iter_num > max_iters:
                         break
-                    if iter_num % args.checkpoint_step == 0:
-                        ckpt_manager.save(epoch, iter_num)
+                if iter_num > max_iters:
```

Comment:
The line are  repeated with line 383.

Reply:
This lines are necessary，as there are three layers loop needing break. Anyway，it is not a elegant way to repeat this lines.